### ISSUE TITLE
Unobvious way to set default application

### ISSUE THREAD

**Issue by jeremypw** [elementary/files#871]
To permanently set the default application for a mime-type through the Files interface, you have to go through the "Other Application ..." submenu of "Open with ..."  Just choosing one of the applications in the "Open With ..." submenu does not set that application as default, which I think is the intended behaviour?

Either that behaviour should be changed, or the wording of the "Other Application ... " menuitem should be altered to indicate that this is the way to set the default application.

Another possibility would be to move the "Set As Default" option into the "Open With ..." submenu (toggling this should not close menu but should be acted on once the application has been chosen)

**Comment by danrabbit** [comment]
I'm not sure that the intention of the user is to change the default handler every time they use "Open With". As an example, the majority of the time I want to open images in Photos but occasionally I may want to open them in GIMP. So I think the existing behavior there is what we want.

A question to ask is if it's within the scope of Files' responsibility to set defaults. We have a basic way to handle common file macro types in System Settings, but there's not really a good way to manage defaults for each individual file type. Perhaps we should work on improving that interface instead?

Thoughts @elementary/ux ?

**Comment by sprite-1** [comment]
I don't agree with making the direct items on Open With automatically be set as the new default handler. The way they have it set up right now is already the optimal way of doing it It's also the standard used across operating systems as far as I know (at least on Windows, it is)

**Comment by jeremypw** [comment]
I agree the behaviour should not be automatic.  The issue is whether the manual method should be more discoverable.

**Comment by jeremypw** [comment]
Closing as no definite conclusion reached.
